### PR TITLE
Fix upload of .archipelago file

### PR DESCRIPTION
**Pull request by zig-for**
## What is this fixing or adding?
Fixes upload of bare `.archipelago` file.
Fixes lack of datapackage stripping from said file.
Fixes broken download button for said upload's patches

## How was this tested?
Uploaded both a .zip and a .archipelago file to a local webhost.

### CODE REVIEW

**Review by alwaysintreble — APPROVED** [review]
LGTM. Tested with 0.3.8 generated game with zip and with just multidata, as well as a seed from source.

File: WebHostLib/upload.py [hunk 1]
```diff
@@ -156,11 +158,11 @@ def uploads():
                     # noinspection PyBroadException
                     try:
                         multidata = file.read()
-                        MultiServer.Context.decompress(multidata)
+                        slots, multidata = parse_multidata(multidata)
                     except Exception as e:
                         flash(f"Could not load multidata. File may be corrupted or incompatible. ({e})")
                     else:
-                        seed = Seed(multidata=multidata, owner=session["_id"])
+                        seed = Seed(multidata=multidata, slots=slots,owner=session["_id"])
```
Comment (black-sliver) [inline WebHostLib/upload.py:165]: Curious. This this ever work without assigning slots?

Also
```suggestion
                        seed = Seed(multidata=multidata, slots=slots, owner=session["_id"])
```

File: WebHostLib/upload.py [hunk 2]
```diff
@@ -20,6 +20,43 @@
 
 banned_zip_contents = (".sfc", ".z64", ".n64", ".sms", ".gb")
 
+def parse_multidata(compressed_multidata, files={}):
+    decompressed_multidata = MultiServer.Context.decompress(compressed_multidata)
+
+    slots: typing.Set[Slot] = set()
+    recompress = False
```
Comment (black-sliver) [inline WebHostLib/upload.py:27]: If you remove the conditional that depends on recompress, please also remove the variable.
```suggestion
```

File: WebHostLib/upload.py [hunk 3]
```diff
@@ -20,6 +20,43 @@
...
+        game_data_packages: typing.List[GameDataPackage] = []
+        for game, game_data in decompressed_multidata["datapackage"].items():
+            if game_data.get("checksum"):
+                game_data_package = GameDataPackage(checksum=game_data["checksum"],
+                                                    data=pickle.dumps(game_data))
+                decompressed_multidata["datapackage"][game] = {
+                    "version": game_data.get("version", 0),
+                    "checksum": game_data["checksum"]
+                }
+                recompress = True
```
Comment (black-sliver) [inline WebHostLib/upload.py:39]: same as above
```suggestion
```

File: WebHostLib/upload.py [hunk 4]
```diff
@@ -20,6 +20,43 @@
 
 banned_zip_contents = (".sfc", ".z64", ".n64", ".sms", ".gb")
 
+def parse_multidata(compressed_multidata, files={}):
```
Comment (black-sliver) [inline WebHostLib/upload.py:23]: Last but not least: a function called `parse_*` writing to the database is not good. Maybe `process_multidata`?

**Review by black-sliver — CHANGES_REQUESTED** [review]

File: WebHostLib/upload.py (hunk 4 above)
Comment (zig-for) [reply WebHostLib/upload.py:23]: done

Comment (black-sliver) [reply WebHostLib/upload.py:23]: did you forget to push?

**Comment by zig-for** [comment]
yes `Warning: the ECDSA host key for 'github.com' differs from the key for
the IP address '192.30.255.113'`

**Review by black-sliver — APPROVED** [review]

**Pull request merged**